Restructured DR backup section for clarity and thoroughness (#20873)
cockroachdb/docs: src/current/v25.4/work-with-virtual-clusters.md

When connected to a virtual cluster from the DB Console, metrics which measure SQL and related activity show data scoped to the virtual cluster. All other metrics are collected system-wide and display the same data on all virtual clusters including the system virtual cluster.

## Backup and restore

> Cockroach Labs recommends that you regularly [back up]({% link {{ page.version.version }}/take-full-and-incremental-backups.md %}#full-backups) your data. When using virtual clusters, perform backups on the _application virtual cluster (app VC)_. Only the app VC's data is included in these backups, and data for other virtual clusters or for the _system virtual cluster (system VC)_ is omitted. You can also back up your system VC to preserve metadata related to your system VC, but this data is usually not critical.

Reviewer: ```suggestion
Cockroach Labs recommends regularly [backing up]({% link {{ page.version.version }}/take-full-and-incremental-backups.md %}#full-backups) your data. When using virtual clusters, perform backups on the _application virtual cluster (app VC)_. Only the app VC's data is included in these backups. Data from other virtual clusters or the _system virtual cluster (system VC)_ is omitted. You can also back up your system VC to preserve its metadata, although this data is usually not critical.
```

When connected to a virtual cluster from the DB Console, metrics which measure SQL and related activity show data scoped to the virtual cluster. All other metrics are collected system-wide and display the same data on all virtual clusters including the system virtual cluster.

## Backup and restore

> Cockroach Labs recommends that you regularly [back up]({% link {{ page.version.version }}/take-full-and-incremental-backups.md %}#full-backups) your data. When using virtual clusters, perform backups on the _application virtual cluster (app VC)_. Only the app VC's data and settings are included in these backups, and data and settings for other virtual clusters or for the _system virtual cluster (system VC)_ are omitted. 

Reviewer 2: nit: do we want to have a final sentence which is kind of saying 'you don't need this data in the system virtual cluster' or like basically reassuring users that the system virtual cluster data is not critical

### Create a backup schedule for your app VC

Cockroach Labs recommends using [backup schedules]({% link {{ page.version.version }}/create-schedule-for-backup.md %}) to automate full and incremental backups of your data.

> Use the following process to create a schedule for a cluster-level backup of your app VC. In this example, the schedule takes revision history for the backup every day at midnight.

Reviewer 3: uber nit: i'd prefer providing a backup example without options (e.g revision history). We don't want users reaching for non default backup flavors because they are in general examples.

> Use the following process to create a schedule for a cluster-level backup of your app VC. In this example, the schedule takes revision history for the backup every day at midnight.

Author: @msbutler what is the default for this? Like if they don't explicitly configure revision history, will no revision history be taken at all? I want to make sure they know what they're getting with the default if that's what we're putting here

    (2 rows)
    ~~~

    {% include {{ page.version.version }}/backups/backup-storage-collision.md %}

For information on scheduling backups at different levels or with other options, consult [CREATE SCHEDULE FOR BACKUP]({% link {{ page.version.version }}/create-schedule-for-backup.md %}).

### Back up your app VC

Use the following process to take a one-off full backup of your app VC.

1. [Connect](#connect-to-a-virtual-cluster) to the app VC as a user with [supported privileges]({% link {{ page.version.version }}/security-reference/authorization.md %}#supported-privileges) on the app VC. In this example the user has the `BACKUP` privilege.

    {% include_cached copy-clipboard.html %}
    ~~~ shell
    cockroach sql --url \
    "postgresql://root@{primary node IP or hostname}:26257?options=-ccluster={app_virtual_cluster_name}&sslmode=verify-full" \
    --certs-dir "certs"
    ~~~

1. [Perform a full backup]({% link {{ page.version.version }}/backup.md %}#back-up-a-cluster):

    {% include_cached copy-clipboard.html %}
    ~~~ sql
    BACKUP INTO 'external://backup_s3/app' AS OF SYSTEM TIME '-10s';
    ~~~

> ### Back up a database from your app VC

Reviewer 3: i also don't think we need additional examples for database and table backups. we can provide a cluster level backup and then point to the backup docs. the flow is the same.

    {% include_cached copy-clipboard.html %}
    ~~~ sql
    BACKUP test_schema.* INTO 'external://backup_s3/app/schema' AS OF SYSTEM TIME '-10s';
    ~~~

### Back up your system VC

You can also back up your system VC to preserve metadata such as users and cluster settings. Use the following process to back up your system VC.

1. [Connect](#connect-to-the-system-virtual-cluster) to the system VC as a user with [supported privileges]({% link {{ page.version.version }}/security-reference/authorization.md %}#supported-privileges) on the system VC. In this example the user has the `BACKUP` privilege.

    {% include_cached copy-clipboard.html %}
    ~~~ shell
    cockroach sql --url \
    "postgresql://root@{primary node IP or hostname}:26257?options=-ccluster=system&sslmode=verify-full" \
    --certs-dir "certs"
    ~~~

1. [Perform a full backup]({% link {{ page.version.version }}/backup.md %}#back-up-a-cluster):

    {% include_cached copy-clipboard.html %}
    ~~~ sql
    BACKUP INTO 'external://backup_s3/system' AS OF SYSTEM TIME '-10s';
    ~~~

    {% include {{ page.version.version }}/backups/backup-storage-collision.md %}

### Restore a virtual cluster

> If needed, you can restore backups to a new app VC with no user-created databases or tables. To restore your app VC from the latest full backup:

Reviewer 2: I think we should give some more examples here - restoring a backup, like a table backup, to the same appVC. Or, restoring a backup to a new appVC which you already have.

Reviewer 2: Yes that sound perfect to me. nothing is different haha good point.

> 1. [Connect to the destination app VC](#connect-to-a-virtual-cluster) as a user with [supported privileges]({% link {{ page.version.version }}/security-reference/authorization.md %}#supported-privileges) on the system VC. In this example the user has the `RESTORE` privilege.

Reviewer 3: nit: "on the system vc" should be "on the app vc"

> When cluster virtualization is enabled, [backup]({% link {{ page.version.version }}/backup.md %}) and [restore]({% link {{ page.version.version }}/restore.md %}) commands are scoped to the virtual cluster by default.

Reviewer 2: +1

Cockroach Labs recommends that you regularly [back up]({% link {{ page.version.version }}/take-full-and-incremental-backups.md %}#full-backups) your _application virtual cluster (app VC)_. Only the app VC's data and settings are included in these backups, and data and settings for other virtual clusters or for the _system virtual cluster (system VC)_ are omitted. If needed, you can [restore](#restore-a-virtual-cluster) these backups to a new app VC. Use the following process to back up your app VC.

1. [Connect](#connect-to-a-virtual-cluster) to the app VC as a user with the `admin` role on the app VC:

    {% include_cached copy-clipboard.html %}
    ~~~ shell
    cockroach sql --url \
    "postgresql://root@{primary node IP or hostname}:26257?options=-ccluster={app_virtual_cluster_name}&sslmode=verify-full" \
    --certs-dir "certs"
    ~~~

> 1. [Perform a full backup]({% link {{ page.version.version }}/backup.md %}#back-up-a-cluster):

Reviewer 3: we recommend that users run backups via backup schedules ([ref](https://www.cockroachlabs.com/docs/stable/create-schedule-for-backup)).  Schedules have a nicer UX (no need to manually take backups), and they manage the backup's protected timestamp.

Reviewer 3: I am curious though, as a new hire, do you think our backups docs should more explicitly point customers to use schedules?

Reviewer 2: I think it would be nice to have both. "Perform a one off full backup or create a backup schedule so that backups can be automatically taken on your behalf at a set frequency" (that wasn't great wording but something along those lines) and then we have a code snippet for setting a backup schedule too.

Reviewer 2: I'm cool with both.

    {% include_cached copy-clipboard.html %}
    ~~~ sql
    BACKUP INTO 'external://backup_s3' AS OF SYSTEM TIME '-10s';
    ~~~

>     {% include {{ page.version.version }}/backups/backup-storage-collision.md %}

Reviewer 2: @msbutler I feel like I barely understand this directive. It reads to me like we don't want to take multiple full backups to the same place. Any ideas on how we could maek this clearer

You can also back up your system VC to preserve metadata such as users and cluster settings. Use the following process to back up your system VC.

> 1. [Connect](#connect-to-the-system-virtual-cluster) to the system VC as a user with the `admin` role on the system VC:

Reviewer 3: nit: same comment about admin role

1. [Connect to the destination virtual cluster](#connect-to-a-virtual-cluster) as a user with the `admin` role on the virtual cluster.
1. [Restore the cluster]({% link {{ page.version.version }}/restore.md %}). Only the virtual cluster's data and settings are restored.

> 

Reviewer 3: I think the section above can also be deleted. it applies to tenant level backups.

```
You can restore a backup of a virtual cluster to:
- The original virtual cluster on the original CockroachDB cluster.
- A different virtual cluster on the original CockroachDB cluster.
- A different virtual cluster on a different CockroachDB cluster with cluster virtualization enabled.

To restore only a virtual cluster:
1. [Connect to the destination virtual cluster](#connect-to-a-virtual-cluster) as a user with the `admin` role on the virtual cluster.
1. [Restore the cluster]({% link {{ page.version.version }}/restore.md %}). Only the virtual cluster's data and settings are restored.
```

> 

Reviewer 3: > We're not telling users to use cluster level backups

 why not?

> 

Reviewer 2: Let's be more specific: by cluster level I mean taking a backup of the whole physical cluster. If we're talking about app vc level, then we should say 'vc level.' 

My point was that we should **not** document your second bullet, where we take a app vc backup from the system vc,  AND we should also **not** document 'taking a backup of their entire cluster' which would be a 'cluster level backup.' I'm referring to the 'include secondary clusters' option when you take a backup
